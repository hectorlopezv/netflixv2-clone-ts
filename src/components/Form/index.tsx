import React from 'react';
import {
    Container,
    Base, 
    Error, 
    Title, 
    Text, 
    SmallText, 
    Link, 
    Input, 
    Submit
} from './styles/Form';


export interface FormProps {

}

const Form: React.FC<FormProps> = ({ children, ...props }) => {
    return (<Container {...props}>{children}</Container>);
}

export default Form;

export interface BaseProps {
    onSubmit?: (event: any) => void;
    method?: string;

}
 
export const Form_Base: React.FC<BaseProps> = ({children, ...props}) => {
    return (<Base {...props}>{children}</Base>);
}
 

export interface Form_ErrorProps {
    
}
 
export const Form_Error: React.FC<Form_ErrorProps> = ({children, ...props}) => {
    return (<Error {...props}>{children}</Error>);
}


export interface Form_TitleProps {
    
}
 
export const Form_Title: React.FC<Form_TitleProps> = ({children, ...props}) => {
    return (  <Title {...props}>{children}</Title>);
}
 

export interface Form_TextProps {
    
}
 
export const Form_Text: React.FC<Form_TextProps> = ({children, ...props}) => {
    return (<Text {...props}>{children}</Text>  );
}
 
export interface Form_SmallTextProps {
    
}
 
export const Form_SmallText: React.FC<Form_SmallTextProps> = ({children, ...props}) => {
    return ( <SmallText {...props}>{children}</SmallText> );
}

export interface Form_LinkProps {
    to: string;
}
 
export const Form_Link: React.FC<Form_LinkProps> = ({children, ...props}) => {
    return ( <Link {...props}>{children}</Link> );
}
 
export interface Form_InputProps {
    placeholder: string;
    value?: string;
    onChange: ({ target: any }: { target: any; }) => void;
    type?: string;
    autoComplete?: string;
}
 
export const Form_Input: React.FC<Form_InputProps> = ({children, ...props}) => {
    return ( <Input {...props}>{children}</Input> );
}
 
export interface Form_SubmitProps {
    disabled: any;
    type: any;
    
}
 
export const Form_Submit: React.FC<Form_SubmitProps> = ({children, ...props}) => {
    return (  <Submit {...props}>{children}</Submit>);
}
 

 



 
