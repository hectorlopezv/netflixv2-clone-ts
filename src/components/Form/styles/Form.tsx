import styled from 'styled-components/macro';

export const Container = styled.div``;
export const Base  = styled.div``;
export const Error = styled.div``; 
export const Title = styled.div``; 
export const Text  = styled.div``;
export const SmallText = styled.div``;
export const Link  = styled.div``;
export const Input = styled.div``; 
export const Submit = styled.div``;