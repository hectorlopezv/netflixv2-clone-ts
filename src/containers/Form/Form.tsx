/* eslint-disable react/jsx-pascal-case */
import React from 'react';
import OptForm, {OptForm_Input, OptForm_Button, OptForm_Text, OptForm_Break} from '../../components/optForm';


export interface FormProps {
    
}
 
const Form: React.FC<FormProps> = () => {
    return (  
        <OptForm>
            <OptForm_Input placeholder="Email Address"/>
            <OptForm_Button>Try it now</OptForm_Button>
            <OptForm_Break/> 
            <OptForm_Text>
                Ready to watch? enter your email 
                to create or restart your membership
            </OptForm_Text>
        </OptForm>
    );
}
export default Form;