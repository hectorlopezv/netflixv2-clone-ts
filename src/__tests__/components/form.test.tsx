/* eslint-disable react/jsx-pascal-case */
import React from 'react';
import { render, fireEvent, getByTestId, getByAltText } from '@testing-library/react';


import Form, {
    Form_Link,
    Form_Input,
    Form_Base,
    Form_Error,
    Form_Title,
    Form_Submit,
    Form_SmallText,
    Form_Text,
} from '../../components/Form';

//mocking Example on libraries or services!
jest.mock('react-router-dom');


describe('<Form />', () => {
    it('renders the <Form /> with populated data', () => {
        const { container, getByText, getByPlaceholderText } = render(
          <Form>
            <Form_Title>Sign In Now</Form_Title>
    
            <Form_Base>
              <Form_Input placeholder="Email address" onChange={() => {}} />
              <Form_Input type="password" placeholder="Password" onChange={() => {}} />

            </Form_Base>


            <Form_SmallText>
              This page is protected by Google reCAPTCHA to ensure you're not a bot. Learn more.
            </Form_SmallText>
          </Form>
        );
    
        expect(
          getByText("This page is protected by Google reCAPTCHA to ensure you're not a bot. Learn more.")
        ).toBeTruthy();
        expect(getByPlaceholderText('Email address')).toBeTruthy();
        expect(getByPlaceholderText('Password')).toBeTruthy();
        expect(container.firstChild).toMatchSnapshot();
      });
});