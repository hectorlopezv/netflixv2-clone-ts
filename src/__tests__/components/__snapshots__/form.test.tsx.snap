// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> renders the <Form /> with populated data 1`] = `
<div
  class="Form__Container-sc-13a4k0z-0 gnOEDv"
>
  <h1
    class="Form__Title-sc-13a4k0z-3 fiTcPU"
  >
    Sign In Now
  </h1>
  <form
    class="Form__Base-sc-13a4k0z-1 QfanM"
  >
    <input
      class="Form__Input-sc-13a4k0z-7 cvttIU"
      placeholder="Email address"
    />
    <input
      class="Form__Input-sc-13a4k0z-7 cvttIU"
      placeholder="Password"
      type="password"
    />
  </form>
  <p
    class="Form__SmallText-sc-13a4k0z-5 BIKtN"
  >
    This page is protected by Google reCAPTCHA to ensure you're not a bot. Learn more.
  </p>
</div>
`;
